src/core/dep-graph.ts

```diff
@@ -161,6 +161,10 @@ class DepGraphImpl implements types.DepGraphInternal {
       otherDepGraph = createFromJSON(other.toJSON()) as types.DepGraphInternal;
     }
 
+    // In theory, for the graphs created by standard means, `_.isEquals(this._data, otherDepGraph._data)`
+    // should suffice, since node IDs will be generated in a predictable way.
+    // However, to support unconventional node IDs, we run our own deep
```

Comment:
not only - it's also to:
* compare across outputs of different (compatible) versions of the lib
* be not sensitive to ordering as the data contains arrays

Reply:
the array ordering should be fairly deterministic, I believe (but that hinges on the current V8 implementation)